src/kvstore/raftex/RaftPart.cpp

```diff
@@ -947,56 +1024,15 @@ void RaftPart::processAppendLogResponses(const AppendLogResponses& resps,
               << lastLogId;
     }
 
-    // Step 4: Fulfill the promise
-    if (iter.hasNonAtomicOpLogs()) {
-      sendingPromise_.setOneSharedValue(nebula::cpp2::ErrorCode::SUCCEEDED);
-    }
-    if (iter.leadByAtomicOp()) {
-      sendingPromise_.setOneSingleValue(nebula::cpp2::ErrorCode::SUCCEEDED);
-    }
-    // Step 5: Check whether need to continue
-    // the log replication
+    // at this monment, we have confidence logs should be succeeded replicated
+    // sendingPromise_.setOneSharedValue(nebula::cpp2::ErrorCode::SUCCEEDED);
     {
       std::lock_guard<std::mutex> lck(logsLock_);
       CHECK(replicatingLogs_);
-      // Continue to process the original AppendLogsIterator if necessary
-      iter.resume();
-      // If no more valid logs to be replicated in iter, create a new one if we
-      // have new log
-      if (iter.empty()) {
-        VLOG(4) << idStr_ << "logs size " << logs_.size();
-        if (logs_.size() > 0) {
-          // continue to replicate the logs
-          sendingPromise_ = std::move(cachingPromise_);
-          cachingPromise_.reset();
-          iter = AppendLogsIterator(firstLogId,
-                                    currTerm,
-                                    std::move(logs_),
-                                    [this](AtomicOp op) -> std::optional<std::string> {
-                                      auto opRet = op();
-                                      if (!opRet.has_value()) {
-                                        // Failed
-                                        sendingPromise_.setOneSingleValue(
-                                            nebula::cpp2::ErrorCode::E_RAFT_ATOMIC_OP_FAILED);
-                                      }
-                                      return opRet;
-                                    });
-          logs_.clear();
-          bufferOverFlow_ = false;
-        }
-        // Reset replicatingLogs_ one of the following is true:
-        // 1. old iter is empty && logs_.size() == 0
-        // 2. old iter is empty && logs_.size() > 0, but all logs in new iter is
-        // atomic op,
-        //    and all of them failed, which would make iter is empty again
-        if (iter.empty()) {
-          replicatingLogs_ = false;
-          VLOG(4) << idStr_ << "No more log to be replicated";
-          return;
-        }
-      }
+      iter.commit();
```

Comment:
Maybe we still need to build another iterator, otherwise we need to wait heartbeat to trigger replication.